### ISSUE TITLE
No method to receive Firebase Message / Notification

### ISSUE THREAD

**Issue by rupakTj** [GitLiveApp/firebase-kotlin-sdk#623]
Firebase Messaging does not provide to receive the message. 

Do not have any method like `override fun onMessageReceived(message: RemoteMessage)` provided by `FirebaseMessagingService`

**Comment by mr-kew** [comment]
I use platform specific code for that. See [#467](https://github.com/GitLiveApp/firebase-kotlin-sdk/issues/467#issuecomment-2110968316)

**Comment by rupakTj** [comment]
Yes, I have used the same [https://search.maven.org/artifact/dev.gitlive/firebase-messaging/2.1.0/pom](Firebase Messaging library), but this lib only contains methods for fetching token, subscribe/unsubscribe topics. 

But there is method to receive message. Am i missing anything.

**Comment by mr-kew** [comment]
I am now not sure if I understood the question correctly.

What i meant is that you can obtain the [native firebase messaging](https://firebase.google.com/docs/cloud-messaging/ios/client) inside you iosMain (for example) by doing:

```
val messaging = Firebase.messaging.ios

// You can then proceed to handle the notifications as you would do in swift.

messaging.delegate = this
```

But this library (even in its native form) is not meant to receive notifications directly. You are just meant to register your app to receive notifications from Firebase. The receiving & displaying itself is handled by the respective os (that means in iosApp).

**Comment by rupakTj** [comment]
thanks @mr-kew for the guidance. Yes in the native layer it can be done, 

but it would have been great for one stop solution if this library can provide the message receive part as well.

**Comment by spporan** [comment]
> I am now not sure if I understood the question correctly.
> 
> What i meant is that you can obtain the [native firebase messaging](https://firebase.google.com/docs/cloud-messaging/ios/client) inside you iosMain (for example) by doing:
> 
> ```
> val messaging = Firebase.messaging.ios
> 
> // You can then proceed to handle the notifications as you would do in swift.
> 
> messaging.delegate = this
> ```
> 
> But this library (even in its native form) is not meant to receive notifications directly. You are just meant to register your app to receive notifications from Firebase. The receiving & displaying itself is handled by the respective os (that means in iosApp).

So Only should we use this library to get token from native side? And other implementation need to be done in native side then why should we use this library?

**Comment by nbransby** [comment]
Cross-referencing: #778 raises this too — "in Messaging there is no class to extend `FirebaseMessagingService` ...hence impossible for receiving notifications" — bundled with an unrelated Storage request. This issue is the more focused one for the Messaging side, and it was filed first.
